### PR TITLE
fix: Fix designkit screenshots being affected by subpixel lengths

### PR DESCRIPTION
**Pull request by ashtonG**
## Description

This fixes issues where adding a new component flags all following components in the designkit as changed due to the elements being shifted unevenly. We introduce an exclusive mode to the design kit that hides all other components on the page when navigating via the nav.

## Test Plan

ci only -- no testing needed

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

WEB-1222

### CODE REVIEW

File: webui/react/src/pages/DesignKit.tsx [hunk 1]
```diff
@@ -2365,17 +2368,17 @@ const DesignKit: React.FC = () => {
...
               </li>
             ))}
           </ul>
         </nav>
         <article>
-          {componentOrder.map((componentId) => (
-            <React.Fragment key={componentId}>{Components[componentId]}</React.Fragment>
-          ))}
+          {componentOrder
+            .filter((id) => !isExclusiveMode || !location.hash || id === location.hash.substring(1))
```
Comment (hkang1) [inline webui/react/src/pages/DesignKit.tsx:2378]: noice

File: webui/react/src/pages/DesignKit.tsx [hunk 2]
```diff
@@ -2365,17 +2368,17 @@ const DesignKit: React.FC = () => {
           <ul>
             {componentOrder.map((componentId) => (
               <li key={componentId}>
-                <Link reloadDocument to={`#${componentId}`}>
-                  {ComponentTitles[componentId]}
-                </Link>
+                <a href={`#${componentId}`}>{ComponentTitles[componentId]}</a>
```
Comment (hkang1) [inline webui/react/src/pages/DesignKit.tsx:2371]: **question:** was this change to make it easier to match the links?
```
const links = await page.locator('nav ul a').all();
```

Comment (hkang1) [reply webui/react/src/pages/DesignKit.tsx:2371]: **question:** when `isExclusiveMode` is true, do we need to preserve it and keep passing `exclusive=true` via other links like these `<a>` links?

**Review by hkang1 — COMMENTED** [review]
just had a couple of questions based on a quick look. Could use someone else's eyes who is familiar with this area. Otherwise I can take another more in depth pass if there isn't someone.

File: webui/react/src/pages/DesignKit.tsx (hunk 2 above)
Comment (ashtonG) [reply webui/react/src/pages/DesignKit.tsx:2371]: We made this change so the `isExclusiveMode` query param is preserved when clicking on links. The `Link` element doesn't keep the search part of the url when navigating to a hash.

As other links aren't used in generating the screenshots and the intent of `exclusive` is mainly to generate screenshots, using the same treatment isn't necessary, but I can update it for consistency's sake.

**Review by johnkim-det — APPROVED** [review]
LGTM, although I'm not sure whether we actually need to scroll through all the sections on one page -- could be fine to just always use "exclusive mode", use hash to determine which section to render, and not worry about managing the search param. But that would take away the ability to ctrl+f search the entire page. just thinking out loud here. I think for the sake of fixing snapshot testing this is good to merge, and we can revisit the page structure later if necessary.

File: webui/react/src/pages/DesignKit.tsx [hunk 3]
```diff
@@ -2365,17 +2336,17 @@ const DesignKit: React.FC = () => {
...
               </li>
             ))}
           </ul>
         </nav>
         <article>
```
[the remark this answers is not in the document]
Comment (ashtonG) [reply webui/react/src/pages/DesignKit.tsx:2344]: let's capture that as cleanup.

**Pull request merged**